MaxText/layers/attentions.py

```diff
@@ -159,22 +143,28 @@ def query_init(*args):
   def kv_projection(self, inputs_kv: Array, proj_name: str) -> Array:
     """Projection for Key and Value.
 
-    Args;
-      inputs_kv: inputs_kv: key/values of shape `[batch, kv_length, kv_features]`.
+    Args:
+      inputs_kv: inputs_kv: key/values of shape `[batch, kv_length,
+        num_kv_heads, kv_dim]`.
       proj_name: name of projection, `key` or `value`.
 
     Returns:
-      Projection of key or value, in shape of `[batch, kv_length, num_heads,
-      head_dim]`.
+      Projection of key or value, in shape of `[batch, kv_length, head_dim]`.
     """
+    if self.num_kv_heads == -1:
+      raise ValueError('num_kv_heads is not defined.')
+
+    if self.num_query_heads % self.num_kv_heads != 0:
+      raise ValueError('Invaid num_kv_heads for GQA.')
```

Comment:
Nit: This error could be made more helpful by showing the user what values are incorrect or why.

Reply:
ack